### PR TITLE
[ray-operator] Watch the pod resource and remove useless code

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: Ce Gao <gaoce@caicloud.io>

## Why are these changes needed?

The function `Add` is generated by kubebuilder v1 scaffolding. And We did not use it. Then the pod resource is not watched by the controller. We should use the function `SetupWithManager` since it is called in the main.go.

PTAL

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: deploy/ray-operator/controllers/raycluster_controller.go [hunk 1]
```diff
@@ -29,45 +28,14 @@ import (
 
 var log = logf.Log.WithName("RayCluster-Controller")
 
-// Add creates a new RayCluster Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
-// and start it when the Manager Started.
-func Add(mgr manager.Manager) error {
```
Comment (gaocegege) [inline deploy/ray-operator/controllers/raycluster_controller.go]: The code is never executed, thus we did not watch the pod resource successfully.

**Comment by gaocegege** [comment]
Hi, admins

How can I restart a new jenkind build?

**Comment by simon-mo** [comment]
Jenkins build only test RLlib, this should be unrelated.

**Review by simon-mo — APPROVED** [review]

**Review by simon-mo — APPROVED** [review]
LGTM, but I will wait for @edoakes to merge

**Comment by gaocegege** [comment]
> Jenkins build only test RLlib, this should be unrelated.

@simon-mo Gotcha, thanks.

> LGTM, but I will wait for @edoakes to merge

Sure, SGTM, thanks for your review.

**Pull request merged**